ports/stm32/uart.c

```diff
@@ -689,7 +691,14 @@ int uart_rx_char(pyb_uart_obj_t *self) {
         self->uartx->ICR = USART_ICR_ORECF; // clear ORE if it was set
         return data;
         #else
-        return self->uartx->DR & self->char_mask;
+        int data = self->uartx->DR & self->char_mask;
+        if (self->mp_irq_trigger & UART_FLAG_RXNE) {
+            self->uartx->CR1 |= USART_CR1_RXNEIE;
```

Comment:
FWIW the `UART_RXNE_IT_EN(self->uartx);` macro was used in the other if block just above this to re-enable the interrupt after stalled flow control - it would be worth making these two scenario's consistent in terms of interrupt handling.